src/rgw/rgw_gc.cc

```diff
@@ -229,7 +230,11 @@ int RGWGC::list(int *index, string& marker, uint32_t max, bool expired_only, std
         marker.clear();
       }
     }
-    if (transitioned_objects_cache[*index] || check_queue) {
+    if (transitioned_objects_cache[*index] || check_queue || processing_queue) {
+      //If processing queue is set to true, then reset it
+      if (processing_queue) {
```

Comment:
this can be outside the if in line 233

Reply:
++

Reply:
lines 264 and 266, are under a special condition where if the number of entries == max, then depending on whether the entries come from queue and the truncated flag, the processing queue flag is set or cleared. Lets say the flag is set here, then in the next iteration, if the number of entries remaining in queue < max, then this flag will not be cleared. And the for loop will go on to the next gc object, and based on the flag being set, it will only process the queue and will not check the omap for entries.

Reply:
ok, isee. then just set it to false here, no need for the "if" in line 235

Reply:
ok, i agree to that.